regexes/device/mobiles.yml

```
@@ -1880,15 +1915,39 @@ Changhong:
 
 # Cherry Mobile
 'Cherry Mobile':
-  regex: 'Cherry|Flare2X|Fusion Bolt|Flare S Play|OMEGA HD 4'
+  regex: 'Cherry|Flare([ _])?(2X|4|A[123]|J[1235]|P[13]|S[456]|X2)|Flare S Play|Flare_HD_MAX|FLARE LITE|Flare XL Plus|Fusion Bolt|OMEGA HD 4'
   device: 'smartphone'
   models:
     - regex: 'Cherry(?: ?Mobile)?[ _]?([^/;]+) Build'
       model: '$1'
     - regex: '(Flare2X)'
       model: '$1'
+    - regex: 'Flare_HD_MAX'
+      model: 'Flare HD Max'
+    - regex: 'FLARE LITE'
+      model: 'Flare Lite'
     - regex: 'Flare S Play'
       model: 'Flare S Play'
+    - regex: 'Flare XL Plus'
+      model: 'Flare XL Plus'
+    - regex: 'Flare[ _](A2|J3|P1|S[46])[ _]Lite'
+      model: 'Flare $1 Lite'
+    - regex: 'Flare[ _](J[12]|S5)[ _]mini'
+      model: 'Flare $1 Mini'
+    - regex: 'Flare J2S'
+      model: 'Flare J2s'
+    - regex: 'Flare S4 Max LTE'
+      model: 'Flare S4 Max LTE'
+    - regex: 'Flare S4 Max'
+      model: 'Flare S4 Max'
+    - regex: 'FlareS5LiteDTV '
+      model: 'Flare S5 Lite DTV'
+    - regex: 'Flare_S6_Plus'
+      model: 'Flare S6 Plus'
+    - regex: 'FLARE S6 POWER'
+      model: 'Flare S6 Power'
+    - regex: 'Flare([ _])?(4|A[13]|J[135]|P[13]|S4|X2)'
```

Not necessarily needed, but using this you could use `$1` below
```suggestion
    - regex: 'Flare(?:[ _])?(4|A[13]|J[135]|P[13]|S4|X2)'
```